tools/clang/lib/SPIRV/DeclResultIdMapper.cpp

```diff
@@ -600,6 +600,52 @@ bool containOnlyVecWithFourFloats(QualType type, bool use16Bit) {
   return false;
 }
 
+// Evaluates whether a given QualType is in fact an array and unroll accordingly.
+// Will return the the appropriate RecordType for the provided QualType. Will be 
+// the actual type of the array element, if it's really an array.
```

Comment:
Can you briefly mention that arraySizes is an output parameter and explain the order of array dimensions?